src/components/Subscriber/Subscriber.tsx

```diff
@@ -0,0 +1,100 @@
+import Input from '@/components/Input';
+import MultiSelect from '@/components/MultiSelect';
+import Button from '@/components/Button';
+import { getTags } from '@/apis/tag';
+import { useEffect, useState } from 'react';
+import { Formik, Form } from 'formik';
+import { subscriberValidationSchema } from '@/validations/subscriber';
+
+const Subscriber = ({
+  onSubmit,
+  selectedOptions,
+  setSelectedOptions,
+  subscriber = null,
+}: {
+  onSubmit: any;
+  selectedOptions: any;
+  setSelectedOptions: any;
+  subscriber: any;
+}) => {
+  const [tags, setTags] = useState([]);
+
+  useEffect(() => {
+    const fetchTagLists = async () => {
+      setTags(await getTags());
+    };
+
+    fetchTagLists();
+  }, []);
+
+  const tagsList =
+    Array.isArray(tags.data) && tags.data.length > 0
+      ? tags.data.map((tag: any) => ({
+          value: tag.id,
+          label: tag.name,
+        }))
```

Comment:
it seems you are formatting tags several times. Once here, and I saw it in edit page also. Do you think we can create a common utility function that can help with transforming the tags in the way we need it and reuse that function instead of code repetition?

Reply:
Please check this [comment](https://github.com/ColoredCow/campaigns-web/pull/3#discussion_r1269881012)